File: 2020 listopad/python/1.py
with open("dane.txt", "r") as f:
    dane = list(map(int, f.read().strip().split('\n')))

lucky = [i for i in range(1, 10001, 2)]
cur = 1
while cur < len(lucky):
    crossed = [lucky[i] for i in range(lucky[cur] - 1, len(lucky), lucky[cur])]
    for i in crossed:
        lucky.remove(i)
    cur += 1
lucky = set(lucky)

ans = 0
for i in dane:
    if i in lucky:
        ans += 1
print(ans)

File: 2020 listopad/c++/1.cpp
#include <iostream>
#include <fstream>
#include <list>
#include <set>
using namespace std;

list <int> lucky;
set <int> s;

int main()
{
    ifstream in("dane.txt");
    
    for (int i = 1; i <= 10000; i += 2)
        lucky.push_back(i);
    
    list <int>::iterator cur = lucky.begin();
    cur++;
    while (cur != lucky.end())
    {
        int l = *cur, i = 1;
        for (list <int>::iterator it = lucky.begin(); it != lucky.end(); it++)
        {
            if (i++ % l == 0)
            {
                list <int>::iterator tmp = --it;
                it++;
                lucky.erase(it);
                it = tmp;
            }
        }
        cur++;
    }

    for (int v: lucky)
        s.insert(v);
    
    int n, ans = 0;
    while (in >> n)
    {
        if (s.count(n))
            ans++;        
    }   
    cout << ans << '\n';
}